src/triggers/messages.ts

```diff
@@ -0,0 +1,42 @@
+
+import { Table } from '../db/db';
+import { CrdtMessage, CrdtOpetation } from '../types/message';
+import { Cursor, Trigger } from '../types/trigger';
+
+export const pendingMempoolMessages: (tables: string) => Trigger<undefined> = 
+  (table) => {
+    return async (clients) => {
+      const sql = `
+      select rm.*, rcs."lastTimestamp"
+      from raw_mempool rm 
+      left outer join ${table} t 
+      on rm."entityId" = t.id
+      left outer join ${Table.crdtState} rcs
+      on rm."table" = rcs."table" 
+      and rm."column" = rcs."column" 
+      and rm."entityId" = rcs."entityId"
+      where rm."table" = '${table}'
+      and ((rm.operation = '${CrdtOpetation.UPDATE}' and t.id is not null) or rm.operation = '${CrdtOpetation.INSERT}')
+      order by rm."table", rm."column", rm."entityId", rm.timestamp
+      `;
+
+      const result = await clients.db.rawSQL(sql);
+      return result.rows;
+    }
+  }
+
+export const newMessages: Trigger<Cursor> = async (clients, cursor: string) => {
+  const selectSql = `select m.* from ${Table.seeds} m
+  limit ${parseInt(process.env.QUERY_TRIGGER_BATCH_SIZE!)}  
+  offset ${parseInt(cursor)}
```

Comment:
maybe we should have an explicity autoincrementing private key and sort by it explicitly instead of just depending on offset?